### PR TITLE
Dataset can be initialized via passing source and target sequence lists

### PR DESCRIPTION
**Pull request by kpfadnis**
Address #4

### CODE REVIEW

File: seq2seq/dataset/dataset.py [hunk 1]
```diff
@@ -28,23 +28,47 @@ class Dataset(object):
         tgt_max_vocab (int): maximum target vocabulary size
     """
 
-    def __init__(self, path, src_max_len, tgt_max_len, src_vocab=None, tgt_vocab=None, src_max_vocab=50000,
-                 tgt_max_vocab=50000):
+    def __init__(self, src_max_len, tgt_max_len):
```
Comment (kylegao91) [inline seq2seq/dataset/dataset.py]: Now that the class docstring does not match `__init__` anymore.

File: seq2seq/dataset/dataset.py [hunk 2]
```diff
@@ -28,23 +28,47 @@ class Dataset(object):
...
+        obj = cls(src_max_len, tgt_max_len)
         pairs = utils.prepare_data(path, src_max_len, tgt_max_len)
 
         # Read in vocabularies
-        self.input_vocab = self._init_vocab(zip(*pairs)[0], src_max_vocab, src_vocab)
-        self.output_vocab = self._init_vocab(zip(*pairs)[1], tgt_max_vocab, tgt_vocab)
+        obj.input_vocab = obj._init_vocab(zip(*pairs)[0], src_max_vocab, src_vocab)
+        obj.output_vocab = obj._init_vocab(zip(*pairs)[1], tgt_max_vocab, tgt_vocab)
+
+        # Translate input sequences to token ids
```
Comment (kylegao91) [inline seq2seq/dataset/dataset.py]: The lines in the second parts of `from_file` and those of `from_list` are duplicated.  Would be better if extract as a function.

File: seq2seq/dataset/dataset.py (hunk 1 above)
Comment (kpfadnis) [inline seq2seq/dataset/dataset.py]: Updating documentation as recommend.

File: seq2seq/dataset/dataset.py (hunk 2 above)
Comment (kpfadnis) [inline seq2seq/dataset/dataset.py]: Changed as recommended.

File: seq2seq/dataset/dataset.py [hunk 3]
```diff
@@ -5,36 +5,111 @@ class Dataset(object):
...
+            tgt_vocab (Vocabulary): pre-populated Vocabulary object or a path of a file containing words for the target language,
+            default `None`. If a pre-populated Vocabulary object, `tgt_max_vocab` wouldn't be used.
+            src_max_vocab (int): maximum source vocabulary size
+            tgt_max_vocab (int): maximum target vocabulary size
+        """
+        obj = cls(src_max_len, tgt_max_len)
+        pairs = utils.prepare_data_from_list(src_data, tgt_data, src_max_len, tgt_max_len)
+        return cls.encode(obj, pairs, src_vocab, tgt_vocab, src_max_vocab, tgt_max_vocab)
+
+    def encode(self, pairs, src_vocab=None, tgt_vocab=None, src_max_vocab=50000, tgt_max_vocab=50000):
```
Comment (kylegao91) [inline seq2seq/dataset/dataset.py]: Is `encode` a user-facing method?  Remember not to show it in the doc, or just add an underscore prefix.

**Pull request merged**